### ISSUE TITLE
Way to specify inclusive namespace prefix for XML canonicalization?

### ISSUE THREAD

**Issue by lzen** [IdentityPython/pysaml2#249]
Hi, I'm new to SAML and just encountered a problem when using pysaml2 to build an IdP. When building a signed request, the canonicalization algorithm specified in xmldsig/**init**.py is exclusive canonicalization ('http://www.w3.org/2001/10/xml-exc-c14n#') and it will not include declarations for some namespaces which are necessary for some SPs (e.g. namespace xmlns:xs for QNames like in xsi:type="xs:string"). The W3C recommendation specified InclusiveNamespaces PrefixList to handle these cases, but I couldn't find how to do it in pysaml2. There's a similar issue here:
https://bz.apache.org/bugzilla/show_bug.cgi?id=49710

Is there a solution? Either include these namespaces, or just remove all these xsi:type="xs:string".

Thank you!

**Comment by lnyntng** [comment]
Hi! Did you manage to find a solution? I am going through the same issue...

**Comment by lnyntng** [comment]
Hi!

I managed to declare the prefixes by declaring them as shown below:

```python
nsprefix={
    'saml2p': 'urn:oasis:names:tc:SAML:2.0:protocol',
    'saml2': 'urn:oasis:names:tc:SAML:2.0:assertion',
    'ds': 'http://www.w3.org/2000/09/xmldsig#',
}
```

**Comment by danielrich** [comment]
Inyntng where did you place that to get it picked up and used?

**Comment by lnyntng** [comment]
I sent it as a parameter in the method `prepare_for_authenticate`

**Comment by c00kiemon5ter** [comment]
This should be resolved by #365 
I am closing this.
